src/discord-cluster-manager/leaderboard_eval.py

```diff
@@ -28,30 +26,28 @@ def metric():
     warmup_runs = 10
     timed_runs = 100
 
-    # warmup
+    # Warmup Code
     print('warming up...')
     for _ in range(warmup_runs):
         input_tensors = generate_input()
-        for input in input_tensors:
-            _ = custom_kernel(input)
-            _ = ref_kernel(input)
+        _ = custom_kernel(input_tensors)
+        _ = ref_kernel(input_tensors)
 
-    # timing
-    print('timing custom implementation...')
-    input_tensor = generate_input()
+    # Timing Code
+    input_tensors = generate_input()
     start_time = time.time()
     for _ in range(timed_runs):
-        for input in input_tensors:
-            _ = custom_kernel(input)
+        _ = custom_kernel(input_tensors)
+    end_time = time.time()
```

Comment:
This is missing a cuda sync, i.e the timings will only be correct if the kernel code ends by doing the cuda sync itself by for example sending tensors back to cpu to print them